src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1442,6 +1443,13 @@ bool InterpreterCreateQuery::doCreateTable(ASTCreateQuery & create,
                         "ATTACH ... FROM ... query is not supported for {} table engine, "
                         "because such tables do not store any data on disk. Use CREATE instead.", res->getName());
 
+    if (getContext()->getSettingsRef().create_replicated_merge_tree_fault_injection)
+    {
+        bool is_replicated_storage = typeid_cast<const StorageReplicatedMergeTree *>(res.get()) != nullptr;
+        if (is_replicated_storage)
+            throw Exception(ErrorCodes::ABORTED, "Shutdown is called for table");
```

Comment:
It's not true, the exception message should tell that a fault was injected. Also it makes sense to throw a KeeperException with `ZCONNECTIONLOSS` code or smth like this (but `ABORTED` may be okay too)